diff --git a/ics/structures/op_eth_general_settings.py b/ics/structures/op_eth_general_settings.py
index 046b3ac..3267e34 100644
--- a/ics/structures/op_eth_general_settings.py
+++ b/ics/structures/op_eth_general_settings.py
@@ -5,11 +5,21 @@ import ctypes
 class flags(ctypes.Structure):
     _pack_ = 2
     _fields_ = [
+        ('bTapEnSwitch', ctypes.c_uint32, 1), # [Bitfield] 
+        ('bTapEnPtp', ctypes.c_uint32, 1), # [Bitfield] 
+        ('bEnReportLinkQuality', ctypes.c_uint32, 1), # [Bitfield] 
     ]
 
 # Extra names go here:
 # End of extra names
 
+class op_eth_general_settings_u1(ctypes.Union):
+    _pack_ = 2
+    _fields_ = [
+        ('flags', flags), 
+        ('uFlags', ctypes.c_uint32), 
+    ]
+
 class op_eth_general_settings(ctypes.Structure):
     _pack_ = 2
     _fields_ = [
@@ -21,11 +31,7 @@ class op_eth_general_settings(ctypes.Structure):
         ('tapPair3', ctypes.c_uint16), 
         ('tapPair4', ctypes.c_uint16), 
         ('tapPair5', ctypes.c_uint16), 
-        ('flags', flags), 
-        ('bTapEnSwitch', ctypes.c_uint32, 1), # [Bitfield] 
-        ('bTapEnPtp', ctypes.c_uint32, 1), # [Bitfield] 
-        ('bEnReportLinkQuality', ctypes.c_uint32, 1), # [Bitfield] 
-        ('uFlags', ctypes.c_uint32), 
+        ('u1', op_eth_general_settings_u1), 
     ]
 
 # Extra names go here:
diff --git a/ics/structures/op_eth_settings.py b/ics/structures/op_eth_settings.py
index d1f85dd..d8a07d0 100644
--- a/ics/structures/op_eth_settings.py
+++ b/ics/structures/op_eth_settings.py
@@ -1,11 +1,9 @@
 # This file was auto generated; Do not modify, if you value your sanity!
 import ctypes
 
-class op_eth_settings(ctypes.Structure):
+class op_eth_settings_u1_s1(ctypes.Structure):
     _pack_ = 2
     _fields_ = [
-        ('ucConfigMode', ctypes.c_uint8), 
-        ('preemption_en', ctypes.c_ubyte), 
         ('mac_addr1', ctypes.c_ubyte * 6), # Original Addr for spoofing
         ('mac_addr2', ctypes.c_ubyte * 6), # Target Addr for spoofing
         ('mac_spoofing_en', ctypes.c_ushort, 1), # [Bitfield] 
@@ -13,9 +11,23 @@ class op_eth_settings(ctypes.Structure):
         ('link_spd', ctypes.c_ushort, 2), # [Bitfield] 
         ('q2112_phy_mode', ctypes.c_ushort, 1), # [Bitfield] 
         ('reserved', ctypes.c_ushort, 11), # [Bitfield] 
+    ]
+
+class op_eth_settings_u1(ctypes.Union):
+    _pack_ = 2
+    _fields_ = [
+        ('s1', op_eth_settings_u1_s1), 
         ('reserved0', ctypes.c_ubyte * 14), 
     ]
 
+class op_eth_settings(ctypes.Structure):
+    _pack_ = 2
+    _fields_ = [
+        ('ucConfigMode', ctypes.c_uint8), 
+        ('preemption_en', ctypes.c_ubyte), 
+        ('u1',op_eth_settings_u1), 
+    ]
+
 # Extra names go here:
 OP_ETH_SETTINGS = op_eth_settings
 # End of extra names
 